app/docs/dev/policies/targetref.md

```diff
@@ -0,0 +1,333 @@
+
+A policy metadata looks like:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ExamplePolicy
+metadata:
+  name: my-policy-name
+  namespace: {{site.default_namespace}}
+spec:
+  ... # spec data specific to the policy kind
+```
+
+By default the policy is created in the `default` mesh.
+You can specify the mesh by using the `kuma.io/mesh` label.
+
+For example:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: ExamplePolicy
+metadata:
+  name: my-policy-name
+  namespace: {{site.default_namespace}}
+  labels:
+    kuma.io/mesh: "my-mesh"
+spec:
+  ... # spec data specific to the policy kind
+```
+
+{% warning %}
+Policies are namespaced scope and currently the namespace must be the one the control-plane is running in `{{site.default_namespace}}` by default.
+{% endwarning %}
+
+{% endtab %}
+{% endtabs %}
+
+### The spec
+
+The spec contains the actual configuration of the policy.
+
+All specs have a **top level `targetRef`** which identifies which proxies this policy applies to.
+In particular, it defines which proxies have their Envoy configuration modified.
+
+Some policies also support further narrowing.
+
+The `spec.to[].targetRef` field defines rules that applies to outgoing traffic of proxies selected by `spec.targetRef`.
+The `spec.from[].targetRef` field defines rules that applies to incoming traffic of proxies selected by `spec.targetRef`.
+
+The actual configuration is defined in a `default` map.
+
+For example:
+
+```yaml
+type: ExamplePolicy
+name: my-example
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: Mesh
+      default: # Configuration that applies to outgoing traffic
+        key: value
+  from:
+    - targetRef:
+        kind: Mesh
+      default: # Configuration that applies to incoming traffic
+        key: value
+```
+
+Some policies are not directional and will not have `to` and `from`.
+For example
+
+```yaml
+type: NonDirectionalPolicy
+name: my-example
+mesh: default
+spec:
+  targetRef:
+    kind: Mesh
+  default:
+    key: value
+```
+
+{% tip %}
+One of the benefits of `targetRef` policies is that the spec is always the same between Kubernetes and Universal.
+
+This means that converting policies between universal and Kubernetes only means rewriting the metadata.
+{% endtip %}
+
+#### Writing a `targetRef`
+
+`targetRef` is a concept borrowed from [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/) its usage is fully defined in [MADR 005](https://github.com/kumahq/kuma/blob/master/docs/madr/decisions/005-policy-matching.md).
+Its goal is to select subsets of proxies with maximum flexibility.
+
+It looks like:
+
+```yaml
+targetRef:
+  kind: Mesh | MeshSubset | MeshService | MeshServiceSubset | MeshGatewayRoute
+  name: "my-name" # For kinds MeshService, MeshServiceSubset and MeshGatewayRoute a name can be defined
```

Comment:
Why can't all of them have a name?

Reply:
Cause for Mesh/MeshSubset you don't have a choice right now (it must be the one that the policy is in) In the future we might relax this but not now